core/src/main/java/org/infinispan/persistence/spi/SegmentedAdvancedLoadWriteStore.java

```diff
@@ -212,41 +163,10 @@ default boolean delete(int segment, Object key) {
     * @param fetchValue    whether the value should be included in the marshalled entry
     * @param fetchMetadata whether the metadata should be included in the marshalled entry
     * @return a publisher that will provide the entries from the store that map to the given segments
-    * @implSpec The default implementation falls back to invoking {@link #publishEntries(Predicate, boolean,
-    * boolean)}.
-    * @deprecated since 10.0, please us {@link #entryPublisher(Predicate, boolean, boolean)} instead.
-    */
-   @Deprecated
-   default Publisher<MarshalledEntry<K, V>> publishEntries(IntSet segments, Predicate<? super K> filter, boolean fetchValue,
-                                                     boolean fetchMetadata) {
-      return publishEntries(filter, fetchValue, fetchMetadata);
-   }
-
-   /**
-    * Publishes all entries from this store.  The given publisher can be used by as many {@link
-    * org.reactivestreams.Subscriber}s as desired. Entries are not retrieved until a given Subscriber requests them from
-    * the {@link org.reactivestreams.Subscription}.
-    * <p>
-    * If <b>fetchMetadata</b> is true this store must guarantee to not return any expired entries.
-    * <p>
-    * The segments here <b>must</b> be adhered to and the entries published must not include any that don't belong to
-    * the provided segments.
-    * <p>
-    * This method is not invoked invoked when the store is not configured to be {@link StoreConfiguration#segmented()}.
-    * {@link StoreConfiguration#segmented()}.
-    *
-    * @param segments      the segments that the keys of the entries must map to. Always non null.
-    * @param filter        a filter on the keys of the entries that if passed will allow the given entry to be returned
-    *                      from the publisher
-    * @param fetchValue    whether the value should be included in the marshalled entry
-    * @param fetchMetadata whether the metadata should be included in the marshalled entry
-    * @return a publisher that will provide the entries from the store that map to the given segments
-    * @implSpec The default implementation falls back to invoking {@link #publishEntries(IntSet, Predicate, boolean,
-    * boolean)}.
     */
    default Publisher<MarshallableEntry<K, V>> entryPublisher(IntSet segments, Predicate<? super K> filter, boolean fetchValue,
                                                              boolean fetchMetadata) {
-      return Flowable.fromPublisher(publishEntries(segments, filter, fetchValue, fetchMetadata));
+      return entryPublisher(filter, fetchValue, fetchMetadata);
```

Comment:
This follows the same principle as all of the other default methods, revert to using non-segmented method if no segmented implementations exist. Should we remove those default implemenations? Or keep them and be consistent?

Reply:
IMO there's no point in having any default implementation, either the store is segmented and it can implement all the methods, or it's not really segmented and it shouldn't implement `SegmentedAdvancedLoadWriteStore`.

Reply:
I agree, unless @wburns has a scenario in mind where the non-segmented defaults are a good fit.

Reply:
Nope, that is exactly what I was going for (remove the default impl).